fastlane/Fastfile

```diff
@@ -154,79 +154,133 @@ platform :android do
+            do_create_release_branch(newVersion: newVersion, releaseNotes: releaseNotes)
+            do_create_and_push_tags(newVersion: newVersion)
+        else
+            if UI.confirm("Are you sure you're happy with this release?\n\nVersion=#{newVersion}\nRelease Notes:\n#{releaseNotes}\n")
+                UI.success "Creating release branch for release/#{newVersion}"
+                do_create_release_branch(newVersion: newVersion, releaseNotes: releaseNotes)
+                if UI.confirm(text:"If you have any other changes to make to the release branch, do them now. Enter `y` when ready to create and push tags")
+                    do_create_and_push_tags(newVersion: newVersion)
+                else
+                    UI.error errorMessageCancelled
+                end
+            else
+                UI.error errorMessageCancelled
+            end
+        end
+    end
 
-          File.open('../app/version/version.properties', 'w') do |file|
-            file.write("VERSION=#{newVersion}")
-          end
+        private_lane :do_create_release_branch do |options|
+            newVersion = options[:newVersion]
+            releaseNotes = options[:releaseNotes]
+            sh "git flow release start #{newVersion}"
 
-          File.open('../app/version/release-notes', 'w') do |file|
-            file.write("""#{releaseNotes}""")
-          end
+            File.open('../app/version/version.properties', 'w') do |file|
+                file.write("VERSION=#{newVersion}")
+            end
 
-          if UI.confirm(text:"If you have any other changes to make to the release branch, do them now. Enter `y` when ready to create and push tags")
+            File.open('../app/version/release-notes', 'w') do |file|
+                file.write("""#{releaseNotes}""")
+            end
+        end
 
+        private_lane :do_create_and_push_tags do |options|
+            newVersion = options[:newVersion]
             git_commit(
                 message: "Updated release notes and version number for new release - #{newVersion}",
                 path: "*",
                 allow_nothing_to_commit: true,
                 skip_git_hooks: true
             )
-            
+
             sh "git flow release finish -mnFS '#{newVersion}' '#{newVersion}'"
             push_git_tags
             sh "git push origin main"
             sh "git push origin develop"
 
             UI.header("#{newVersion} tag has been successfully created. 🎉")
-
-          else
-            UI.error errorMessageCancelled
-          end
-
-        else
-          UI.error errorMessageCancelled
-        end
-
         end
 
+        private_lane :determine_release_notes do |options|
+            notes_type = options[:notes_type]
+            custom_release_notes = options[:release_notes]
+            existingReleaseNotes = File.read(releaseNotesFileBody)
+
+            # This maps to the old flow when no options are being passed to determine_release_notes
```

Comment:
"old flow" is perhaps misleading; i think it's less of an old vs new type scenario, and more that we want to support both options.